### ISSUE TITLE
False Positive: Variable is redundant

### ISSUE THREAD

**Issue by cebe** [kalessil/phpinspectionsea#1670]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 3.0.10 |
| Language level | PHP 7.0

Current behaviour

```php
    /**
     * @param $modelName
     * @return string
     */
    public static function getModelTitel_byModelName($modelName)
    {
        $modelClass = self::getModelClass_byModelName($modelName);
        return $modelClass::getModelTitel();
    }
```

In the above code, `$modelClass` is reported as redundant variable assignment, however it is not possible to use it as a static binding without assigning it first.

![Bildschirmfoto von »2020-12-15 11-12-50«](https://user-images.githubusercontent.com/189796/102201510-84f49400-3ec6-11eb-950e-1bfa816a7602.png)

**Comment by cebe** [comment]
hm...  I just tried the auto fix and it seems such a construct is now valid in PHP 7 (very sure it was not in PHP 5).

```php
        return self::getModelClass_byModelName($modelName)::getModelTitel();
```

Not sure what to do about this. The code after the fix looks very confusing to me, so I'd argue having the variable assignment is much better than the "fix". But feel free to close the issue if you disagree...

**Comment by ea-inspections-team** [comment]
Thanks for reporting @cebe! Fixed!
